Mathlib/Algebra/Module/LinearMap.lean

```diff
@@ -1014,15 +1014,41 @@ end SMul
 
 section Module
 
-variable [Semiring S] [Module S M₂] [SMulCommClass R₂ S M₂]
+variable [Semiring S] [Module S M₂]
 
-instance module : Module S (M →ₛₗ[σ₁₂] M₂) where
+instance module  [SMulCommClass R₂ S M₂] : Module S (M →ₛₗ[σ₁₂] M₂) where
   add_smul _ _ _ := ext fun _ ↦ add_smul _ _ _
   zero_smul _ := ext fun _ ↦ zero_smul _ _
 
-instance [NoZeroSMulDivisors S M₂] : NoZeroSMulDivisors S (M →ₛₗ[σ₁₂] M₂) :=
+instance [NoZeroSMulDivisors S M₂]  [SMulCommClass R₂ S M₂] :
+    NoZeroSMulDivisors S (M →ₛₗ[σ₁₂] M₂) :=
   coe_injective.noZeroSMulDivisors _ rfl coe_smul
 
+/-- If `M` is an `(R, S)` bimodule and `M₂` an `S`-module, then `M ⟶ M₂` is also
+  a bimodule-/
+protected def bimodule
+    [Module R M] [Module S M] [SMulCommClass R S M] [SMulCommClass R R M] [Module S M₂] :
+    Module R (M →ₗ[S] M₂) where
+  smul r l :=
+  { toFun := fun x => l <| r • x
+    map_add' := fun x y => by dsimp; rw [smul_add, map_add]
+    map_smul' := fun s x => by dsimp; rw [smul_comm, map_smul] }
+  one_smul l := LinearMap.ext fun x => show l _ = _ by rw [one_smul]
+  mul_smul r₁ r₂ l := LinearMap.ext fun x => show l _ = l _ by rw [smul_comm, mul_smul]
+  smul_zero r := rfl
+  smul_add r l₁ l₂ := LinearMap.ext fun x => show (l₁ + _) _ = _ by
+    rw [LinearMap.add_apply, LinearMap.add_apply]; rfl
+  add_smul r₁ r₂ l := LinearMap.ext fun x => show l _ = l _ + l _ by rw [add_smul, map_add]
+  zero_smul l := LinearMap.ext fun x => show l _ = 0 by rw [zero_smul, map_zero]
```

Comment:
Do we lose some generality by assuming `[SMulCommClass R R M]`?

```lean
example [Semiring R] [AddCommMonoid M] [Module R M] [SMulCommClass R R M] (m : M) (r₁ r₂ : R) :
    (r₁ * r₂) • m = (r₂ * r₁) • m := by
  rw [mul_smul, smul_comm, ← mul_smul]
```

If you don't want to assume `CommRing R`, this may be a case where one needs to bite the bullet and use `MulOpposite` to distinguish the left and right actions. For example (just enough to make it work, but perhaps not the best choice):

```suggestion
open MulOpposite in
/-- If `M` is an `(R, S)` bimodule and `M₂` an `S`-module, then `M ⟶ M₂` is also
  a bimodule-/
protected def bimodule
    [Module R M] [Module S M] [SMulCommClass R S M] [Module S M₂] :
    Module Rᵐᵒᵖ (M →ₗ[S] M₂) where
  smul r l :=
  { toFun := fun x => l (r.unop • x)
    map_add' := fun x y => by dsimp; rw [smul_add, map_add]
    map_smul' := fun s x => by dsimp; rw [smul_comm, map_smul] }
  one_smul l := LinearMap.ext fun x => show l _ = _ by rw [unop_one, one_smul]
  mul_smul r₁ r₂ l := LinearMap.ext fun x => show l _ = l _ by rw [unop_mul, mul_smul]
  smul_zero r := rfl
  smul_add r l₁ l₂ := LinearMap.ext fun x => show (l₁ + _) _ = _ by
    rw [LinearMap.add_apply, LinearMap.add_apply]; rfl
  add_smul r₁ r₂ l := LinearMap.ext fun x => show l _ = l _ + l _ by
    rw [unop_add, add_smul, map_add]
  zero_smul l := LinearMap.ext fun x => show l _ = 0 by rw [unop_zero, zero_smul, map_zero]
```

Not assuming either `CommRing R` or `SMulCommClass R R M` here makes things difficult in "TensorHomAdjunction.lean", of course. I won't try to add `ᵐᵒᵖ` there, since it depends on your decision here.

Reply:
Thanks for the suggestion